src/utils/validation/index.ts

```diff
@@ -42,3 +42,35 @@ export function isValidQuantity(quantity: string) {
 
   return true;
 }
+
+// further validation for sending functions, this is the last line of defense for
+// bad values finding their way into the "private" methods which require paying JUP
+// such as order placement, sending JUP, sending assets, etc...
+export function sendValidation(value: string): boolean {
+  console.log("validating quantity used for send:", value);
+
+  // run the original validation function again as a sanity check
+  if (!isValidQuantity(value)) {
+    return false;
+  }
+
+  // cannot be negative
+  if (parseInt(value) < 0) {
+    console.error("value cannot be negative");
+    return false;
+  }
+
+  // maximum possible value for JUP should be total supply
+  if (parseInt(value) > 1000000000 ** PrecisionExponent) {
```

Comment:
1000000000 seems like a magic number related to the total supply. perhaps a good idea to put this into constants?

especially if supply can ever change?

Reply:
done